api/logs.go

```diff
@@ -61,11 +61,17 @@ func (api *LogsAPI) GetLogs(ctx context.Context, crit FilterQuery) ([]*logType,
 	params := []interface{}{}
 	var goHeavy bool
 	if crit.BlockHash != nil {
-		var num int64
+		var num sql.NullInt64
 		api.db.QueryRowContext(ctx, "SELECT number FROM blocks WHERE hash = ?", trimPrefix(crit.BlockHash.Bytes())).Scan(&num)
 		whereClause = append(whereClause, "blockHash = ? AND block = ?")
-		goHeavy = (num == 0)
-		params = append(params, trimPrefix(crit.BlockHash.Bytes()), num)
+		if !num.Valid { 
+			goHeavy = true 
+		} else {
+			params = append(params, trimPrefix(crit.BlockHash.Bytes()), num.Int64)
```

Comment:
The extension to whereClause needs to happen in conjunction with the extension to params. There should be no scenario where params gets updated without whereclause getting updated, or there will be a mismatch in the number of arguments when the query executes (which happens despite goHeavy being true)